src/collection_manager.cpp

```diff
@@ -528,9 +528,36 @@ Option<nlohmann::json> CollectionManager::drop_collection(const std::string& col
 
     s_lock.unlock();
 
+    auto embedding_fields = collection->get_embedding_fields();
```

Comment:
Should be before `s_lock.unlock()` if you want to copy the fields, or move it after `u_lock` below if you want to have reference. I recommend the second approach (reference).